compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3612,7 +3612,7 @@ boolean validateFloatLiteral(Location pos, String numericLiteral) {
 
     boolean isByteLiteralValue(Long longObject) {
 
-        return (longObject.intValue() >= BBYTE_MIN_VALUE && longObject.intValue() <= BBYTE_MAX_VALUE);
+        return (longObject >= BBYTE_MIN_VALUE && longObject <= BBYTE_MAX_VALUE);
```

Comment:
https://github.com/ballerina-platform/ballerina-lang/issues/32687 This issue has been fixed with https://github.com/ballerina-platform/ballerina-lang/pull/34669 PR. But the same issue with `byte` seems to have not been fixed. Therefore added the same fix for `byte` as well.